Aztec/Classes/Public/TextKit/TextAttachment.swift

```diff
@@ -116,6 +129,43 @@ public class TextAttachment: NSTextAttachment
 
         return CGRect(origin: CGPointZero, size: CGSize(width: width, height: onScreenHeight(width)))
     }
+
+    func updateImage() {
+        switch kind {
+        case .RemoteImage:
+            break
+        default:
+            return
+        }
+        guard !isFetchingImage else {
+            return
+        }
+        isFetchingImage = true
+
+        let imageURL: NSURL
+        switch kind {
+        case .RemoteImage(let url):
+            imageURL = url
+        default:
+            return
+        }
+
+        image = imageProvider?.image(forURL: imageURL, inAttachment: self,
+                                                onSuccess: { [weak self](image) in
+                                                    self?.isFetchingImage = false
+                                                    self?.image = image
+                                                    self?.kind = .RemoteImageDownloaded(url: imageURL, image: image)
+                                                    self?.triggerUpdate()
+            }, onFailure: { [weak self]() in
+                self?.isFetchingImage = false
+                self?.kind = .MissingImage
+                self?.triggerUpdate()
+            })        
+    }
+
+    func triggerUpdate(){
+        self.textContainer?.layoutManager?.invalidateLayoutForAttachment(self)
```

Comment:
IMHO the text attachment should not know the layout manager at all.

This should be executed automatically by our `TextStorage` [here](https://github.com/wordpress-mobile/WordPress-Aztec-iOS/pull/114/files#diff-b08a98a1418f1c0ea7d9ae88558e1aaaR478).

Then [this property](https://github.com/wordpress-mobile/WordPress-Aztec-iOS/pull/114/files#diff-ac4b458a2b7aba8c3f8e9aad368bae1dR31) would also go away.